util/failover.go

```diff
@@ -316,3 +302,69 @@ func GetPreferredNodes(clientset *kubernetes.Clientset, selector, namespace stri
 
 	return nodes, err
 }
+
+// ToggleAutoFailover enables or disables autofailover for a cluster.  Disabling autofailover means "pausing"
+// Patroni, which will result in Patroni stepping aside from managing the cluster.  This will effectively cause
+// Patroni to stop responding to failures or other database activities, e.g. it will not attempt to start the
+// database when stopped to perform maintenance
+func ToggleAutoFailover(clientset *kubernetes.Clientset, enable bool, clusterName, namespace string) error  {
+	
+	// find the "config" configMap created by Patroni
+	configMapName := clusterName+"-config"
+	log.Debugf("setting autofailover to %t for cluster %s", enable, clusterName)
+
+	configMap, found := kubeapi.GetConfigMap(clientset, configMapName, namespace)
+	if found {
```

Comment:
Could we 'short-circuit' here? It would at least reduce on level of nested conditionals.

Something like this perhaps?

```
if !found {
  log(err)  
  return err
}

... other stuff ...
```